.changeset/gold-impalas-retire.md

```diff
@@ -0,0 +1,5 @@
+---
+"@coinbase/onchainkit": minor
+---
+
+Replaced internal `useOnchainActionWithCache` with `tanstack/react-query`. This affects `useName` and `useAvatar` hooks. The return type and the input parameters also changed for these 2 hooks.
```

Comment:
Nice!!!!

Also more big nit, if you want you can split this message in multiple parts and add `- **feat**` and get more into details on the why and what.

But again big nit.

Reply:
Actually now that I think about it.

Because we export
```
export { useAvatar } from './identity/hooks/useAvatar';
export { useName } from './identity/hooks/useName';
```

we need to explain the breaking changes here as well.

Reply:
oh good point. Will do